From: Oleksij Rempel <o.rempel@pengutronix.de>
Date: Wed, 6 Oct 2021 10:43:23 +0200
Subject: [PATCH] ARM: boards: skov-imx6: use separate DTS for the iMX6 Solo
 variant

iMX6 Solo boards do not have HDMI so remove it to avoid time spending on
probing.

Signed-off-by: Oleksij Rempel <o.rempel@pengutronix.de>
Link: https://lore.barebox.org/20211006084323.14051-9-o.rempel@pengutronix.de
Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>
---
 arch/arm/boards/skov-imx6/lowlevel.c |  6 +++++-
 arch/arm/dts/Makefile                |  2 +-
 arch/arm/dts/imx6s-skov-imx6.dts     | 22 ++++++++++++++++++++++
 3 files changed, 28 insertions(+), 2 deletions(-)
 create mode 100644 arch/arm/dts/imx6s-skov-imx6.dts

diff --git a/arch/arm/boards/skov-imx6/lowlevel.c b/arch/arm/boards/skov-imx6/lowlevel.c
index eab797faa1dc..ea6de36a36c1 100644
--- a/arch/arm/boards/skov-imx6/lowlevel.c
+++ b/arch/arm/boards/skov-imx6/lowlevel.c
@@ -618,6 +618,7 @@ static void skov_imx6_init(int cpu_type, unsigned board_variant)
 
 extern char __dtb_z_imx6q_skov_imx6_start[];
 extern char __dtb_z_imx6dl_skov_imx6_start[];
+extern char __dtb_z_imx6s_skov_imx6_start[];
 
 /* called twice: once for SDRAM setup only, second for devicetree setup */
 static noinline void skov_imx6_start(void)
@@ -640,8 +641,11 @@ static noinline void skov_imx6_start(void)
 	/* boot this platform (second call) */
 	switch (cpu_type) {
 	case IMX6_CPUTYPE_IMX6S:
+		pr_debug("Startup i.MX6S based system...\n");
+		imx6q_barebox_entry(__dtb_z_imx6s_skov_imx6_start);
+		break;
 	case IMX6_CPUTYPE_IMX6DL:
-		pr_debug("Startup i.MX6S/DL based system...\n");
+		pr_debug("Startup i.MX6DL based system...\n");
 		imx6q_barebox_entry(__dtb_z_imx6dl_skov_imx6_start);
 		break;
 	case IMX6_CPUTYPE_IMX6D:
diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 80caa738a9b8..88a079497a36 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -115,7 +115,7 @@ lwl-$(CONFIG_MACH_SOLIDRUN_CUBOX) += dove-cubox-bb.dtb.o
 lwl-$(CONFIG_MACH_SOLIDRUN_MICROSOM) += imx6dl-hummingboard.dtb.o imx6q-hummingboard.dtb.o \
 				imx6dl-hummingboard2.dtb.o imx6q-hummingboard2.dtb.o \
 				imx6q-h100.dtb.o
-lwl-$(CONFIG_MACH_SKOV_IMX6) += imx6dl-skov-imx6.dtb.o imx6q-skov-imx6.dtb.o
+lwl-$(CONFIG_MACH_SKOV_IMX6) += imx6s-skov-imx6.dtb.o imx6dl-skov-imx6.dtb.o imx6q-skov-imx6.dtb.o
 lwl-$(CONFIG_MACH_SKOV_ARM9CPU) += at91-skov-arm9cpu.dtb.o
 lwl-$(CONFIG_MACH_SEEED_ODYSSEY) += stm32mp157c-odyssey.dtb.o
 lwl-$(CONFIG_MACH_STM32MP15XX_DKX) += stm32mp157c-dk2.dtb.o stm32mp157a-dk1.dtb.o
diff --git a/arch/arm/dts/imx6s-skov-imx6.dts b/arch/arm/dts/imx6s-skov-imx6.dts
new file mode 100644
index 000000000000..9061563e4593
--- /dev/null
+++ b/arch/arm/dts/imx6s-skov-imx6.dts
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright 2015 Juergen Borleis, Pengutronix <kernel@pengutronix.de>
+ */
+
+/dts-v1/;
+#include <arm/imx6dl.dtsi>
+#include "imx6dl.dtsi"
+#include "imx6qdl-skov-imx6.dtsi"
+
+/ {
+	model = "Skov IMX6";
+	compatible = "skov,imx6", "fsl,imx6dl";
+
+	chosen {
+		stdout-path = &uart2;
+	};
+};
+
+&hdmi {
+	status = "disabled";
+};
